india_forecast_app/app.py

```diff
@@ -1,10 +1,184 @@
+"""
+Main forecast app entrypoint
+"""
+
+import datetime as dt
+import logging
+import os
+import sys
+
 import click
+import pandas as pd
+from pvsite_datamodel import DatabaseConnection
+from pvsite_datamodel.read import get_sites_by_country
+from pvsite_datamodel.write import insert_forecast_values
+from sqlalchemy.orm import Session
+
+from .model import DummyModel
+
+log = logging.getLogger(__name__)
+
+
+def get_site_ids(db_session: Session) -> list[str]:
+    """
+    Gets all avaiable site_ids in India
+    
+    Args:
+            db_session: A SQLAlchemy session
+
+    Returns:
+            A list of site_ids
+    """
+    
+    sites = get_sites_by_country(db_session, country="india")
+
+    return [s.site_uuid for s in sites]
+
+
+def get_model():
+    """
+    Instantiates and returns the forecast model ready for running inference
+
+    Returns:
+            A forecasting model
+    """
+
+    model = DummyModel()
+    return model
+
+
+def run_model(model, site_id: str, timestamp: dt.datetime):
+    """
+    Runs inference on model for the given site & timestamp
+
+    Args:
+            model: A forecasting model
+            site_id: A specific site ID
+            timestamp: timestamp to run a forecast for
+
+    Returns:
+            A forecast or None if model inference fails
+    """
+
+    try:
+        forecast = model.predict(site_id=site_id, timestamp=timestamp)
+    except Exception:
+        log.error(
+            f"Error while running model.predict for site_id={site_id}. Skipping",
+            exc_info=True,
+        )
+        return None
+
+    return forecast
+
+
+def save_forecast(db_session: Session, forecast, write_to_db: bool):
```

Comment:
this could be moved to a different file say `save.py`, this could be done a different PR

Reply:
I'll probably look to refactor all the functions apart from app to other file(s) once I hook this up to the real model.